psycopg3/transaction.py

```diff
@@ -13,35 +13,73 @@
 if TYPE_CHECKING:
     from .connection import Connection
 
+_log = logging.getLogger(__name__)
+
+
+class Rollback(Exception):
+    """
+    Exit the current Transaction context immediately and rollback any changes
+    made within this context.
+
+    If a transaction context is specified in the constructor, rollback
+    enclosing transactions contexts up to and including the one specified.
+    """
+
+    def __init__(self, transaction: "Transaction" = None) -> None:
+        self.transaction = transaction
+
+    def __str__(self) -> str:
+        return f"<Rollback({self.transaction})>"
+
 
 class Transaction:
-    def __init__(self, conn: "Connection") -> None:
+    def __init__(
+        self, conn: "Connection", savepoint_name: str, force_rollback: bool
+    ) -> None:
         self._conn = conn
-        self._outer_transaction = False
         self._savepoint_name: Optional[bytes] = None
+        if savepoint_name is not None:
+            self._savepoint_name = savepoint_name.encode("ascii")
+        self.force_rollback = force_rollback
+
+        self._outer_transaction: bool
 
     @property
     def connection(self) -> "Connection":
         return self._conn
 
+    @property
+    def savepoint_name(self) -> Optional[str]:
+        if self._savepoint_name is None:
+            return None
+        return self._savepoint_name.decode("ascii")
+
+    @property
+    def rollback_exception(self) -> Rollback:
```

Comment:
Looking at tests and use cases, in the test suite you had to import `Rollback` in order to use it, or the user can use this helper method. I think we can do without it and expose the exception in the transaction class, so that the user can do `raise my_tx.Rollback`, `raise my_tx.Rollback(my_tx)`, `raise my_tx.Rollback(other_tx)`.

Reply:
I've not come across this pattern of exposing "relevant types" on another type, with the intent for them to be accessed via instances. What's the problem with importing more types as needed?

I worry that this will introduce extra confusion, particularly in statements like `raise my_tx.Rollback(other_tx)` where two transactions are mentioned but only one is actually relevant.

Also, I don't think it's nice if you have to alias the transaction just for the sake of getting access to the `Rollback` type.

e.g. I prefer this...
```
from psycopg3.transactions import Rollback

with conn.transaction():
    raise Rollback()
```

...rather than this:
```
with conn.transaction() as tx:
    raise tx.Rollback()
```

...and this I think becomes catastrophically confusing:
```
with conn.transaction() as tx1:
    with conn.transaction() as tx2:
        raise tx1.Rollback()
```

I'd propose removing this `rollback_exeption` property -- I don't think it's adding much value and just gives you two ways to do the same thing. `Rollback(tx)` isn't exactly a hardship (however we settle on accessing the `Rollback` type.)

Reply:
If it's confusing that's ok. My problem is with exposing the `psycopg3.transaction` module only to expose `Rollback`. I would rather import it in the top `psycopg3` package. And yes, we can probably do without `rollback_exeption`.

Reply:
Agree that we don't want people importing `psycopg3.transaction` (hence why I initially wanted to _underscore it :-))
I think we should expose `Rollback` in psycopg3.__all__ -- do you agree?

Reply:
I think it's a pretty important object: I wouldn't mind it in the top-level package.